inlong-sort/sort-connectors/filesystem/src/main/java/org/apache/inlong/sort/filesystem/stream/AbstractStreamingWriter.java

```diff
@@ -141,14 +161,10 @@ protected void commitUpToCheckpoint(long checkpointId) throws Exception {
             if (sinkMetricData != null) {
                 sinkMetricData.invoke(rowSize, dataSize);
             }
-        } catch (Exception e) {
-            if (sinkMetricData != null) {
-                sinkMetricData.invokeDirty(rowSize, dataSize);
-            }
-            LOG.error("fileSystem sink commitUpToCheckpoint.", e);
-        } finally {
             rowSize = 0L;
             dataSize = 0L;
+        } catch (Exception e) {
+            LOG.error("fileSystem sink commitUpToCheckpoint.", e);
```

Comment:
here metric data will not be consistent when commit fail

Reply:
done